### PR TITLE
Added Course, university and document API's

### PR DESCRIPTION
**Pull request by arundotrs**
Added Course, University and Document API's , Test is also included. @joelmjohn , Please let me know your comments.

### CODE REVIEW

File: server/controllers/course.controller.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    try {
+        const { id } = req.params;
+        console.log("id is", id)
+        const courseExists = await courseServices.exists(id);
+        console.log("Course", courseExists)
+        if(!courseExists){
+            responseUtil.throwError(MessageUtil.somethingWentWrong);
+        } else {
+            const response = await courseServices.modifyCourse(id, req.body)
+            console.log(response)
```
Comment (joelmjohn) [inline server/controllers/course.controller.js:84]: keep in mind to remove console logs, once you feel its working as expected.

File: server/controllers/course.controller.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        responseUtil.errorResponse(res, err.message);
+    }
+};
+
+exports.removeCourse = async (req, res) => {
+    try {
+        const { id } = req.params;
+        console.log("id is", id)
+        const courseExists = await courseServices.exists(id);
+        console.log("Course", courseExists)
```
Comment (joelmjohn) [inline server/controllers/course.controller.js:101]: remove console log

File: server/services/course.service.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
+const courseModel = require('../models/course.model');
+
+exports.save = async (data) => {
+    const doc = new courseModel(data);
+    return await doc.save();
```
Comment (joelmjohn) [inline server/services/course.service.js:5]: `course` instead of `doc`

File: server/services/university.service.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
+const universityModel = require('../models/university.model');
+
+exports.save = async (data) => {
+    const doc = new universityModel(data);
+    return await doc.save();
```
Comment (joelmjohn) [inline server/services/university.service.js:5]: `doc` -> `university` or `newModel` can be used on every other service.

File: server/services/university.service.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+};
+
+exports.findAllUniversities = async ({page, limit}) => {
+    const mongoQuery = [
+        {$project: { __v: 0, _id: 0}},
+        { $skip: (page - 1) * limit },
+        { $limit: limit }
+    ]
+    return await universityModel.aggregate(mongoQuery)
+};
```
Comment (joelmjohn) [inline server/services/university.service.js:38]: Nice!!

**Review by joelmjohn — APPROVED** [review]
Looks good to me, please merge.

File: server/controllers/course.controller.js (hunk 1 above)
Comment (arundotrs) [reply server/controllers/course.controller.js:84]: Noted.

**Comment by joelmjohn** [comment]
Also it would be good if you could remove make type referencing ID of `document` schema to `string` instead of `type: Schema.Types.ObjectId,`
https://github.com/joelmjohn/foreignmentor-uk/blob/0fd48674516911852797752a0108db7af7282aeb/server/models/document.model.js#L15-L20

**Pull request merged**